Index: hostapd-10.0.0/src/drivers/driver_nl80211.h
===================================================================
--- hostapd-10.0.0.orig/src/drivers/driver_nl80211.h
+++ hostapd-10.0.0/src/drivers/driver_nl80211.h
@@ -95,6 +95,7 @@ struct i802_bss {
 	u64 wdev_id;
 	char ifname[IFNAMSIZ + 1];
 	char brname[IFNAMSIZ];
+	u32 radio_mask;
 	unsigned int added_if_into_bridge:1;
 	unsigned int already_in_bridge:1;
 	unsigned int added_bridge:1;
@@ -346,7 +347,7 @@ send_and_recv_resp(struct wpa_driver_nl8
 
 int nl80211_create_iface(struct wpa_driver_nl80211_data *drv,
 			 const char *ifname, enum nl80211_iftype iftype,
-			 const u8 *addr, int wds,
+			 const u8 *addr, int wds, u32 radio_mask,
 			 int (*handler)(struct nl_msg *, void *),
 			 void *arg, int use_existing);
 void nl80211_remove_iface(struct wpa_driver_nl80211_data *drv, int ifidx);
Index: hostapd-10.0.0/src/drivers/driver_nl80211.c
===================================================================
--- hostapd-10.0.0.orig/src/drivers/driver_nl80211.c
+++ hostapd-10.0.0/src/drivers/driver_nl80211.c
@@ -182,6 +182,8 @@ static void add_ifidx(struct wpa_driver_
 		      int ifidx_reason);
 static void del_ifidx(struct wpa_driver_nl80211_data *drv, int ifidx,
 		      int ifidx_reason);
+static int have_ifidx(struct wpa_driver_nl80211_data *drv, int ifidx,
+		      int ifidx_reason);
 
 static int nl80211_set_channel(struct i802_bss *bss,
 			       struct hostapd_freq_params *freq, int set_chan);
@@ -6708,7 +6710,8 @@ static void nl80211_teardown_ap(struct i
 	}
 
 	nl80211_put_wiphy_data_ap(bss);
-	wpa_driver_nl80211_del_beacon_all(bss);
+	if (bss->flink)
+		bss->flink->beacon_set = 0;
 }
 
 
@@ -8881,7 +8884,7 @@ static void add_ifidx(struct wpa_driver_
 	wpa_printf(MSG_DEBUG,
 		   "nl80211: Add own interface ifindex %d (ifidx_reason %d)",
 		   ifidx, ifidx_reason);
-	if (nl80211_has_ifidx(drv, ifidx, ifidx_reason)) {
+	if (have_ifidx(drv, ifidx, ifidx_reason)) {
 		wpa_printf(MSG_DEBUG, "nl80211: ifindex %d already in the list",
 			   ifidx);
 		return;
@@ -8941,7 +8944,7 @@ static void del_ifidx(struct wpa_driver_
 }
 
 
-int nl80211_has_ifidx(struct wpa_driver_nl80211_data *drv, int ifidx,
+static int have_ifidx(struct wpa_driver_nl80211_data *drv, int ifidx,
 		      int ifidx_reason)
 {
 	int i;
@@ -9050,7 +9053,7 @@ static void handle_eapol(int sock, void
 		return;
 	}
 
-	if (nl80211_has_ifidx(drv, lladdr.sll_ifindex, IFIDX_ANY)) {
+	if (have_ifidx(drv, lladdr.sll_ifindex, IFIDX_ANY)) {
 		for (bss = drv->first_bss; bss; bss = bss->next)
 			drv_event_eapol_rx(bss->ctx, lladdr.sll_addr, buf, len);
 	}
@@ -9358,7 +9361,8 @@ static int wpa_driver_nl80211_if_add(voi
 
 		os_memset(&nonnetdev_info, 0, sizeof(nonnetdev_info));
 		ifidx = nl80211_create_iface(drv, ifname, nlmode, addr,
-					     0, nl80211_wdev_handler,
+					     0, bss->radio_mask,
+					     nl80211_wdev_handler,
 					     &nonnetdev_info, use_existing);
 		if (!nonnetdev_info.wdev_id_set || ifidx != 0) {
 			wpa_printf(MSG_ERROR,
@@ -9377,7 +9381,8 @@ static int wpa_driver_nl80211_if_add(voi
 			   (long long unsigned int) nonnetdev_info.wdev_id);
 	} else {
 		ifidx = nl80211_create_iface(drv, ifname, nlmode, addr,
-					     0, NULL, NULL, use_existing);
+					     0, bss->radio_mask,
+					     NULL, NULL, use_existing);
 		if (use_existing && ifidx == -ENFILE) {
 			added = 0;
 			ifidx = if_nametoindex(ifname);
@@ -9561,9 +9566,6 @@ static int wpa_driver_nl80211_if_remove(
 			wpa_printf(MSG_INFO, "nl80211: %s - could not find "
 				   "BSS %p in the list", __func__, bss);
 	} else {
-		if (eloop_cancel_timeout(wpa_driver_nl80211_scan_timeout, drv, bss->ctx))
-			wpa_driver_nl80211_scan_timeout(drv, bss->ctx);
-
 		wpa_printf(MSG_DEBUG, "nl80211: First BSS - reassign context");
 		nl80211_teardown_ap(bss);
 		nl80211_remove_links(bss);
Index: hostapd-10.0.0/src/drivers/driver_nl80211_monitor.c
===================================================================
--- hostapd-10.0.0.orig/src/drivers/driver_nl80211_monitor.c
+++ hostapd-10.0.0/src/drivers/driver_nl80211_monitor.c
@@ -98,7 +98,7 @@ static int nl80211_create_monitor_interf
 	buf[IFNAMSIZ - 1] = '\0';
 
 	*ifidx = nl80211_create_iface(drv, buf, NL80211_IFTYPE_MONITOR, NULL,
-				      0, NULL, NULL, 0);
+				      0, 0, NULL, NULL, 0);
 
 	if (*ifidx < 0)
 		return -1;
