Index: hostapd-10.0.0/src/ap/sta_info.c
===================================================================
--- hostapd-10.0.0.orig/src/ap/sta_info.c
+++ hostapd-10.0.0/src/ap/sta_info.c
@@ -268,6 +268,15 @@ void clear_wpa_sm_for_all_sta(struct hos
 
 #endif /* CONFIG_IEEE80211BE */
 
+#ifndef CONFIG_IEEE80211BE
+void clear_wpa_sm_for_all_sta(struct hostapd_data *hapd,
+			      struct wpa_state_machine *wpa_sm)
+{
+	(void) hapd;
+	(void) wpa_sm;
+}
+#endif /* !CONFIG_IEEE80211BE */
+
 
 void ap_free_sta(struct hostapd_data *hapd, struct sta_info *sta)
 {
@@ -2176,3 +2185,22 @@ void ap_sta_free_sta_profile(struct mld_
 	}
 }
 #endif /* CONFIG_IEEE80211BE */
+
+bool ap_sta_is_mld(struct hostapd_data *hapd,
+		   struct sta_info *sta)
+{
+#ifdef CONFIG_IEEE80211BE
+	return hapd->conf->mld_ap && sta && sta->mld_info.mld_sta;
+#else /* CONFIG_IEEE80211BE */
+	return false;
+#endif /* CONFIG_IEEE80211BE */
+}
+
+void ap_sta_set_mld(struct sta_info *sta, bool mld)
+{
+#ifdef CONFIG_IEEE80211BE
+	if (sta)
+		sta->mld_info.mld_sta = mld;
+#endif /* CONFIG_IEEE80211BE */
+}
+
Index: hostapd-10.0.0/src/ap/sta_info.h
===================================================================
--- hostapd-10.0.0.orig/src/ap/sta_info.h
+++ hostapd-10.0.0/src/ap/sta_info.h
@@ -419,23 +419,8 @@ int ap_sta_re_add(struct hostapd_data *h
 
 void ap_free_sta_pasn(struct hostapd_data *hapd, struct sta_info *sta);
 
-static inline bool ap_sta_is_mld(struct hostapd_data *hapd,
-				 struct sta_info *sta)
-{
-#ifdef CONFIG_IEEE80211BE
-	return hapd->conf->mld_ap && sta && sta->mld_info.mld_sta;
-#else /* CONFIG_IEEE80211BE */
-	return false;
-#endif /* CONFIG_IEEE80211BE */
-}
-
-static inline void ap_sta_set_mld(struct sta_info *sta, bool mld)
-{
-#ifdef CONFIG_IEEE80211BE
-	if (sta)
-		sta->mld_info.mld_sta = mld;
-#endif /* CONFIG_IEEE80211BE */
-}
+bool ap_sta_is_mld(struct hostapd_data *hapd, struct sta_info *sta);
+void ap_sta_set_mld(struct sta_info *sta, bool mld);
 
 void ap_sta_free_sta_profile(struct mld_info *info);
 
Index: hostapd-10.0.0/src/ap/wpa_auth.c
===================================================================
--- hostapd-10.0.0.orig/src/ap/wpa_auth.c
+++ hostapd-10.0.0/src/ap/wpa_auth.c
@@ -7974,6 +7974,14 @@ struct wpa_group * wpa_select_vlan_wpa_g
 }
 #endif /* CONFIG_IEEE80211BE */
 
+#ifndef CONFIG_IEEE80211BE
+struct wpa_group * wpa_select_vlan_wpa_group(struct wpa_group *gsm, int vlan_id)
+{
+	(void) vlan_id;
+	return gsm;
+}
+#endif /* !CONFIG_IEEE80211BE */
+
 
 void wpa_auth_set_sae_pw_id(struct wpa_state_machine *sm,
 			    const struct wpabuf *pw_id,
